gitpod-io/gitpod
# Add support for JetBrains EAP IDEs (2022.3)

## Description

Now that IDEA/PyCharm/PhpStorm/GoLand _latest_ version is v2022.3 in Gitpod, we can add support for JetBrains IDEs v2022.3.

All code changes were done to accommodate the API differences between v2022.2 and v2022.3. There were no logic/flow changes in the plugin.

The dev server script (`launch-dev-server.sh`) has been improved to allow us locally test EAP and Stable versions of the IDE, which is a common point of failure during Werft builds - because there we build the plugin for EAP and Stable IDE versions simultaneously.
    - By default, running `launch-dev-server.sh`, will launch the JetBrains EAP version.
    - If we want to build launch JetBrains Stable version we run it with the additional 's' parameter: `launch-dev-server.sh -s`

## How to test

1. Access the Preview Environment of this PR
2. Open a workspace using the **Stable** Version of IntelliJ IDEA and check if the Control Center is being displayed correctly and that the IDE shows no errors about Gitpod Plugin.
3. Do the same as _Step 2_, but now using the **Latest** Stable Version of IntelliJ IDEA.

Note: Preview environments are not working properly today. If it's not working when you're testing it, please follow these instructions to test it from a production workspace:

1. Having the Stable IntelliJ IDEA selected as IDE, access <https://gitpod.io/#referrer:jetbrains-gateway:intellij/https://github.com/gitpod-io/gitpod/pull/13400>
2. On terminal, run: `cd components/ide/jetbrains/backend-plugin/`
3. Then run: `./launch-dev-server.sh -s` to launch the **Stable** Version of IntelliJ IDEA and check if the Control Center is being displayed correctly and that the IDE shows no errors about Gitpod Plugin.
    <img width="509" alt="image" src="https://user-images.githubusercontent.com/418083/192833594-5e8c5020-b7be-4c29-a263-8a40caa9b5c8.png">
4. Stop the command with Control+C, and run `rm -rf /workspace/ide-backend`
5. Now run: `./launch-dev-server.sh` to launch the **Latest** Stable Version of IntelliJ IDEA, and do the same checks from _Step 3_.
    <img width="503" alt="image" src="https://user-images.githubusercontent.com/418083/192834469-c805200a-31c4-4e48-8774-099857440d03.png">

## Release Notes

```release-note
Gitpod Plugin was updated to work with the JetBrains IDEs v2022.3 (EAP).
```

## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`
